src/main/java/Domain/Product.java

```diff
@@ -29,6 +31,17 @@ public double getPrice() {
         return _price;
     }
 
+    public void purchaseProduct() {
+        if (_quantity == 0) {
+            throw new ItemOutOfStockExepction("Product is out of stock");
```

Comment:
add log info pls to inform the update of the _quantity of the product

Reply:
Done.